cypress-io/cypress
# fix: change network request sweep interval from 1s -> 10s

- Related https://github.com/cypress-io/cypress/issues/22868 (contains multiple issues, addresses some)

### User facing changelog


Addresses issue where projects loading a large number of JS modules load very slowly. It might also impact other types of files, I am not sure.

### Additional details


A lot of context written here: https://github.com/cypress-io/cypress/issues/22868

In https://github.com/cypress-io/cypress/pull/22462 a refactor was introduced in the part of the code base where we handle pre-requests. It also introduced some code that would periodically clear out requests that were never matched up (matching up refers to matching a request that is intercepted by our browser extension and the same request that comes into our HTTP proxy). We shipped this in Cypress 10.3. 

The initial PR had a bug that caused some requests would be delayed for 500ms. That was fixed in https://github.com/cypress-io/cypress/pull/23227, which was in Cypress 10.5.

I believe there was *another* bug which flew under the radar. The sweep logic was supposed to be run every 10 seconds, according to https://github.com/cypress-io/cypress/pull/22462, but if you look at the code, it's a `setInterval` triggered every 1 second (500ms * 2). 

I changed this to be every 10 seconds, and it fixed the regression. I included a simple app you can use as a minimal example of the regression: https://github.com/lmiller1990/lots-of-modules (recordings below).

My understanding is the 1s sweep was too fast - the JS module request took a while to come in (more than 1s). Having said that, I don't understand why that causes a bottleneck - but you can see the weird behavior in the first recording below, where the JS modules are processed in "chunks". **I don't understand why this is happening**.

## Recording 1: Sweep Interval 1s. JS modules loading slowly, tests time out.

Notice there is a weird "loading in chunks" behavior when you look at the terminal. 

https://user-images.githubusercontent.com/19196536/208371600-5450ad2b-4f05-478f-b6ba-1fac2d96e579.mp4

## Recording 2: Sweep Interval 10s, test does not time out.

Notice there is no "loading in chunks" behavior.


https://user-images.githubusercontent.com/19196536/208377880-7ba0fa55-00bb-4043-879a-31ca55bd7c23.mp4



## Recording 3: Cypress 9

The test runs in 9s - as opposed to the 11s Cypress 11 shows. This is pretty consistent. I am not sure if this is in the network layer, server, or elsewhere, or a bit of both. It's slower, though, which isn't really ideal.

https://user-images.githubusercontent.com/19196536/208371994-a0be6fac-5487-4d9f-8143-99178c5f0df7.mp4


### Steps to test


Not sure on the best way to test this. I can look into it once we decide if we are happy with this fix, though. If you want to experience the difference, you could grab this repo https://github.com/lmiller1990/lots-of-modules and see the instructions in the README to start it up. Then, you can try playing with the sweep interval (just grab this branch and change variable).


### How has the user experience changed?


Loads apps that import lots of JS modules more quickly.

### PR Tasks
